vexpress/sunxi refactoring

jforissier opened the pull request:
Re-factoring vexpress/sunxi platforms to kill duplicate code and prepare for HiKey support.

pascal-brand-st-dev commented:
Having a new pm repository is a bit overkill. Can we have the PM files in kernel or tee repository?

jenswi-linaro commented:
I agree with Pascal.
Nice to see stuff move out from `plat-*` though.

jforissier commented:
Comments addressed. I have dropped the original commits since there the code is small enough to not require incremental review I think. Also note that the last commit also touches plat-stm (I have updated the commit line).

pascal-brand-st-dev commented on core/arch/arm/include/kernel/pm_stubs.h:
```diff
@@ -25,13 +25,13 @@
 
-#ifndef PM_DEBUG_H
-#define PM_DEBUG_H
+#ifndef PM_H
```
PM_STUBS_H instead if PM_H

pascal-brand-st-dev commented on core/arch/arm/include/kernel/pm_stubs.h:
```diff
@@ -25,13 +25,13 @@
-#ifndef PM_DEBUG_H
-#define PM_DEBUG_H
+#ifndef PM_H
+#define PM_H
```
PM_STUBS_H instead of PM_H

pascal-brand-st-dev commented:
2nd commit message: Can't it be

    "Move PM handler stubs to core/arch/arm/kernel/p……m_stubs.c"

instead of

    "stm, sunxi, vexpress: move PM handler stubs to core/arch/arm/kernel/p……m_stubs.c"?

as it is related to all the supported platforms?

jforissier commented:
@pascal-brand-st-dev true, will fix. In my mind we already have hikey ;)

pascal-brand-st-dev commented:
Once PM_H is fixed:

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

jforissier commented:
Review comments have been addressed.

jenswi-linaro commented:
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

jforissier commented:
Review commits squashed, tags applied, commit comment updated.
